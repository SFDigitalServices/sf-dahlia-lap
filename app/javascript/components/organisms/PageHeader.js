import React from 'react'
import PropTypes from 'prop-types'
import BreadCrumbs from '../atoms/BreadCrumbs'

import { map, has, isArray } from 'lodash'

const DefaultAction = ({ action }) => (
  <a href={action.link} className='alt-caps'>
    {action.title}
  </a>
)

const Actions = ({ actions }) => {
  if (actions) {
    const cleanedActions = isArray(actions) ? actions : [actions]
    const actionsList = map(cleanedActions, (action, idx) =>
      has(action, 'title') ? <DefaultAction action={action} key={idx} /> : action
    )

    return (
      <div className='medium-4 columns no-padding'>
        <span className='lead-header_secondary-action'>{actionsList}</span>
      </div>
    )
  } else {
    return null
  }
}

const PageHeader = ({ title, content, action, breadcrumbs, background }) => {
  return (
    <header
      className={`lead-header short ${breadcrumbs ? 'has-breadcrumbs' : ''} bg-${background}`}
    >
      <div className='row full-width inner--3x'>
        <div className='large-12 columns'>
          {breadcrumbs && <BreadCrumbs items={breadcrumbs} />}
          <hgroup className='lead-header_group'>
            <h1 className={`lead-header_title ${breadcrumbs ? 'small' : 'small-serif'} c-oil`}>
              {title}
            </h1>
            {content && (
              <div className='medium-8 columns no-padding'>
                <p>{content}</p>
              </div>
            )}
            <Actions actions={action} />
          </hgroup>
        </div>
      </div>
    </header>
  )
}

PageHeader.defaultProps = {
  background: 'vapor'
}

PageHeader.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  action: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  breadcrumbs: PropTypes.array
}

export default PageHeader
