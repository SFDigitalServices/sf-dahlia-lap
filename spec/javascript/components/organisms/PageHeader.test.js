import React from 'react'
import renderer from 'react-test-renderer'
import PageHeader from 'components/organisms/PageHeader'

describe('PageHeader', () => {
  const title = 'title1'
  const content = 'content1'
  const breadcrumbs = [
    { title: 'Alpha', link: '/alpah' },
    { title: 'Beta', link: '/beta' },
    { title: 'Gama', link: '/gama' },
    { title: 'Pi', link: '/pi' }
  ]
  const action = { link: '/export', title: 'Export' }

  test('should render succesfully', () => {
    const component = renderer.create(
      <PageHeader title={title} content={content} action={action} breadcrumbs={breadcrumbs} />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render succesfully without BreadCrumbs', () => {
    const component = renderer.create(
      <PageHeader title={title} content={content} action={action} />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render succesfully without action', () => {
    const component = renderer.create(
      <PageHeader title={title} content={content} breadcrumbs={breadcrumbs} />
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
