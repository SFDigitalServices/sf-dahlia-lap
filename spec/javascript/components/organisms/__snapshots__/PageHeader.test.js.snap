// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader should render succesfully 1`] = `
<header
  className="lead-header short has-breadcrumbs bg-vapor"
>
  <div
    className="row full-width inner--3x"
  >
    <div
      className="large-12 columns"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          className="breadcrumbs"
        >
          <li>
            <a
              href="/alpah"
            >
              Alpha
            </a>
          </li>
          <li>
            <a
              href="/beta"
            >
              Beta
            </a>
          </li>
          <li>
            <a
              href="/gama"
            >
              Gama
            </a>
          </li>
          <li
            className="current"
          >
            <a
              aria-current="page"
              href="/pi"
            >
              Pi
            </a>
          </li>
        </ol>
      </nav>
      <hgroup
        className="lead-header_group"
      >
        <h1
          className="lead-header_title small c-oil"
        >
          title1
        </h1>
        <div
          className="medium-8 columns no-padding"
        >
          <p>
            content1
          </p>
        </div>
        <div
          className="medium-4 columns no-padding"
        >
          <span
            className="lead-header_secondary-action"
          >
            <a
              className="alt-caps"
              href="/export"
            >
              Export
            </a>
          </span>
        </div>
      </hgroup>
    </div>
  </div>
</header>
`;

exports[`PageHeader should render succesfully without BreadCrumbs 1`] = `
<header
  className="lead-header short has-breadcrumbs bg-vapor"
>
  <div
    className="row full-width inner--3x"
  >
    <div
      className="large-12 columns"
    >
      <hgroup
        className="lead-header_group"
      >
        <h1
          className="lead-header_title small-serif c-oil"
        >
          title1
        </h1>
        <div
          className="medium-8 columns no-padding"
        >
          <p>
            content1
          </p>
        </div>
        <div
          className="medium-4 columns no-padding"
        >
          <span
            className="lead-header_secondary-action"
          >
            <a
              className="alt-caps"
              href="/export"
            >
              Export
            </a>
          </span>
        </div>
      </hgroup>
    </div>
  </div>
</header>
`;

exports[`PageHeader should render succesfully without action 1`] = `
<header
  className="lead-header short has-breadcrumbs bg-vapor"
>
  <div
    className="row full-width inner--3x"
  >
    <div
      className="large-12 columns"
    >
      <nav
        aria-label="breadcrumb"
      >
        <ol
          className="breadcrumbs"
        >
          <li>
            <a
              href="/alpah"
            >
              Alpha
            </a>
          </li>
          <li>
            <a
              href="/beta"
            >
              Beta
            </a>
          </li>
          <li>
            <a
              href="/gama"
            >
              Gama
            </a>
          </li>
          <li
            className="current"
          >
            <a
              aria-current="page"
              href="/pi"
            >
              Pi
            </a>
          </li>
        </ol>
      </nav>
      <hgroup
        className="lead-header_group"
      >
        <h1
          className="lead-header_title small c-oil"
        >
          title1
        </h1>
        <div
          className="medium-8 columns no-padding"
        >
          <p>
            content1
          </p>
        </div>
        <div
          className="medium-4 columns no-padding"
        />
      </hgroup>
    </div>
  </div>
</header>
`;
